### PR TITLE
Add drscl and tests, and fix the dlamch parameters while here.

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
The tests reach all three code paths in the for loop.

File: native/general.go [hunk 1]
```diff
@@ -78,18 +78,15 @@ func max(a, b int) int {
 // fixed values. Probably a way to get them as constants.
 // TODO(btracey): Is there a better way to find the smallest number such that 1+E > 1
 
-var dlamchE, dlamchS, dlamchP float64
+var (
+	// dlamchE is the machine epsilon. For IEEE this is 2^-53.
+	dlamchE = math.Float64frombits(0x3ca0000000000000)
```
Comment (kortschak) [inline native/general.go]: It's a pity the binary exponent form was removed from the language.

File: native/drscl.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	cden := a
+	cnum := 1.0
+	smlnum := dlamchS
+	bignum := 1 / smlnum
+	for {
+		cden1 := cden * smlnum
+		cnum1 := cnum / bignum
+		var mul float64
+		var done bool
+		if math.Abs(cden1) > math.Abs(cnum) && cnum != 0 {
```
Comment (kortschak) [inline native/drscl.go]: Make this whole block a switch and move cnum != 0 to the front of this case.

**Comment by kortschak** [comment]
LGTM with minor comments.

File: native/general.go (hunk 1 above)
Comment (btracey) [inline native/general.go]: This is the first I've heard about it ever being in the language.

File: native/drscl.go (hunk 2 above)
Comment (btracey) [inline native/drscl.go]: Done.

File: native/general.go (hunk 1 above)
Comment (kortschak) [inline native/general.go]: It's not anymore. It was removed for 1.5. It's used by the compiler and not specified, so they removed it.

**Pull request merged**